client/lib/site/index.js

```diff
@@ -100,6 +100,16 @@ Site.prototype.replace = function( attributes ) {
 
 	let changed = false;
 
+	// Set attrs which will get computed and will mean the same before doing equality checks
+	// so the 'change' event is not triggered. See Site.prototype.updateComputedAttributes.
+	if ( attributes.hasOwnProperty( 'options' ) ) {
+		let options = attributes.options;
+
+		if ( ! options.hasOwnProperty( 'default_post_format' ) || options.default_post_format === '0' ) {
+			options.default_post_format = 'standard';
+		}
+	}
+
```

Comment:
I had a nicer implementation ready with `isEqualWith` and an equality customizer:

```js
const computedAttrsCustomizer = ( newAttrVal, siteAttrVal, attrKey ) => {
	// options.default_post_format
	if ( 
		attrKey === 'default_post_format' &&
		newAttrVal === '0' &&
		siteAttrVal === 'standard'
	) {
		return true;
	}

	// Use the default comparison mechanism.
	return;
};
```

However, it didn't work because if the attributes coming from the API doesn't include `options.default_post_format` (which they sometimes don't), the above `if` condition is not run (=> there is no `newAttrVal`) and `isEqualWith` assumes that it's indeed not equal and replaces the `site.options` with `attrs.options` and `changed = true`. 

That's why I had to write the above solution. Can you think of something better?